### ISSUE TITLE
📢 Community Help Wanted 📢 - Help QA the new & improved Tutorial!

### ISSUE THREAD

**Issue by agiannelli** [redwoodjs/learn.redwoodjs.com#203]
## The Redwood v1 Launch :rocket: is fast approaching and we need your help

@cannikin has revamped both Tutorial I & II for v1 RC and we need you, our amazing community members, to help QA the new & improved version to ensure the Tutorial is as amazing as ever. :sunglasses:

## :zap: TL;DR

- :construction: Go through Tutorial I & II on [learn.redwoodjs.com 3](http://learn.redwoodjs.com/)
use the [sample repo](https://github.com/redwoodjs/redwood-tutorial) to Quick Start on Tutorial II
- :exclamation:If you find something…
  - :loudspeaker: [Open an issue](https://github.com/redwoodjs/learn.redwoodjs.com/issues/new)
  - :arrow_heading_down: [Open a PR 1](https://github.com/redwoodjs/learn.redwoodjs.com/compare) (We love [contributors](https://redwoodjs.com/docs/contributing-overview)! :heart: )
- :question: Questions? Reach out here, in the [forums](https://community.redwoodjs.com/t/community-help-wanted-help-qa-the-new-improved-tutorial/2825) :fountain_pen:, or on [Discord](https://discord.gg/redwoodjs) :speech_balloon:

## :question:What we need from you

Head over to [learn.redwoodjs.com]()[](https://community.redwoodjs.com/t/community-help-wanted-help-qa-the-new-improved-tutorial/2825#types-of-tests-were-looking-for-4):arrow_upper_right: and start the Tutorial!

As you work your way through make note of typos :fountain_pen:, sections that are confusing :thinking:, instructions or code that do not work as expected 🙅 , bugs you find :lady_beetle: - any & all hiccups you run into.

We want to know about it. If you’re struggling, chances are someone else is too.

## :test_tube: Types of tests we’re looking for:

- :hash: Different Node versions (Redwood is compatible with v14 → v16)
- :desktop_computer: Operating Systems - Windows, Mac, Linux
- :closed_book: Walk through Tutorial I
- :green_book: Walk through Tutorial II based on self-built Tutorial I
- :blue_book: Tutorial II based on the [sample repo](https://github.com/redwoodjs/redwood-tutorial)[](https://community.redwoodjs.com/t/community-help-wanted-help-qa-the-new-improved-tutorial/2825#reporting-an-issue-5) (The sample repo is the final state of Tutorial I)
 
## :exclamation: Reporting an Issue

If you find something, you can report below ⬇️, [open an issue](https://github.com/redwoodjs/learn.redwoodjs.com/issues/new):rotating_light: against the [learn.redwoodjs.com](https://github.com/redwoodjs/learn.redwoodjs.com) repository or feel free to [open a PR 1](https://github.com/redwoodjs/learn.redwoodjs.com/compare):arrow_heading_down: if you want to fix it yourself. We love [contributors](https://redwoodjs.com/docs/contributing-overview)[](https://community.redwoodjs.com/t/community-help-wanted-help-qa-the-new-improved-tutorial/2825#final-thoughts-6). :heart:

Be sure to include the following in your report:

- :hash: Node Version
- :desktop_computer: OS
- :keyboard: any commands that failed
- :x: error messages
- :camera_flash: screenshots
- :open_book: which part of the Tutorial you were working on
- etc…

## :thought_balloon: Final Thoughts
Feel free to reach out to core team members on this issue, in the [forums](https://community.redwoodjs.com/t/community-help-wanted-help-qa-the-new-improved-tutorial/2825) :fountain_pen:, or on [Discord](https://discord.gg/redwoodjs) :speech_balloon: if you need any assistance.

Go forth and help make Redwood even better. :woman_technologist:t2::man_technologist::rocket:

>By helping each other be successful with Redwood, we make the Redwood project successful. :heart:

**Comment by adriatic** [comment]
I strongly agree on all your points @agiannelli and look forward to a well synchronized and managed tutorial QA under your guidance. My firm opinion is that a nearly perfect tutorial is the key attraction to acquisition of new users / customers., So thank you for doing this ❤️ 

One more question if I may: can you please suggest some information on how to write the tests in the context of this QA?. I found a few like

- [Let’s discuss a command to create a functional test project](https://community.redwoodjs.com/t/lets-discuss-a-command-to-create-a-functional-test-project/2054). _This one discusses a slightly different subject, but some of the ideas there could be relevant here._

- [Add script to create a functional test project using the latest CRWA template](https://github.com/redwoodjs/redwood/pull/2324#issue-862677919) _Same comments as above_

 - It might be wise to solicit the final set of opinions from @dthyresson  (like his feedback on my post [here](https://community.redwoodjs.com/t/community-help-wanted-help-qa-the-new-improved-tutorial/2825/5?u=adriatic) and [here](https://community.redwoodjs.com/t/community-help-wanted-help-qa-the-new-improved-tutorial/2825/8?u=adriatic). I find David's approach to the meaningfulness of testing tutorial (not doing QA, mind you) very close to my own, so it would be very beneficial if you can collect a merged opinion of @dthyresson, @cannikin and @thedavidprice to serve as a part of your guidance to QA.

**Comment by adriatic** [comment]
@thedavidprice - I took on myself to suggest another round of testing issues aggregation - please nix it if I am too late for this (do not to cause troubles to @agiannelli's leadership 😿)

**Comment by palante** [comment]
Insightful tutorial! Created a PR for the `/building-a-form` page, regarding the contact code highlights! https://github.com/redwoodjs/learn.redwoodjs.com/pull/205

**Comment by thedavidprice** [comment]
@palante Awesome. Thank you!

**Comment by rkmitra1** [comment]
Added a tutorial issue (dbAuth not working on Safari) to https://github.com/redwoodjs/redwood/issues/4710#issuecomment-1065206897. Not sure where I should have put my notes but is seems related to the that issue.

**Comment by adriatic** [comment]
@rkmitra1 having the same issue as you (not having the definition of how to report my findings), I created my own repository (https://github.com/adriatic/tutorial-qa) where I am taking detailed notes of my QA findings, so that someone from the core team can look at my notes if necessary. 

I would suggest this same approach to all folks participating in this [QA task](https://community.redwoodjs.com/t/community-help-wanted-help-qa-the-new-improved-tutorial/2825)

Copy @thedavidprice so he can eventually advertise my approach

**Comment by agiannelli** [comment]
@adriatic apologies for my delay in response. I appreciate your support on this initiative. I apologize if my outline was not clear. 

The expectations for this Tutorial QA are: 

- For grammar / spelling issues - users should [open a PR](https://github.com/redwoodjs/learn.redwoodjs.com/compare) to fix or [create an issue](https://github.com/redwoodjs/learn.redwoodjs.com/issues/new) outlining the mistakes.
- For anything broken or not working as expected, an [issue should be created](https://github.com/redwoodjs/learn.redwoodjs.com/issues/new) for review 
- For Clarity / overall experience issues, users can outline and report on this issue or report on [the forum post](https://community.redwoodjs.com/t/community-help-wanted-help-qa-the-new-improved-tutorial/2825). 

In general, users should feel comfortable (and are encouraged!) to open a PR for anything they feel comfortable with fixing themselves, or create an issue to report any bugs.

CC @rkmitra1 @thedavidprice

**Comment by adriatic** [comment]
> In general, users should feel comfortable (and are encouraged!) to open a PR for anything they feel comfortable with fixing themselves, or create an issue to report any bugs.

Re-reading this single sentence I am now puzzled what else did I need 😕  ( at least I cannot resurrect the state of mind at the time I wrote the above text). I do remember how I wanted to provide a lot richer feedback that include the complete source code (presenting my own repo) to ease the problem reproduction. 

Today, I am very with your summary above, so, I would only suggest to provide the output of the command `yarn rw info` with each report that can benefit from such information:
```
nik$ yarn rw info
(node:3351) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

  System:
    OS: macOS 12.2.1
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - /private/var/folders/y3/n2p741pj7vq9ckj9jrmlbp6m0000gn/T/xfs-553be7b3/node
    Yarn: 3.2.0 - /private/var/folders/y3/n2p741pj7vq9ckj9jrmlbp6m0000gn/T/xfs-553be7b3/yarn
  Databases:
    SQLite: 3.36.0 - /usr/bin/sqlite3
  Browsers:
    Chrome: 99.0.4844.51
    Firefox: 96.0.3
    Safari: 15.3
  npmPackages:
    @redwoodjs/core: ^0.49.1 => 0.49.1 
```

Thank you for restating your objectives ❤️
